### PR TITLE
Orbit UI rework: Adds icons, health bars, better view.

### LINKED ISSUE #7521: Refresh button on the Follow TGUI menu does not work
### Description

As per title, it does not refresh the menu once you press the button.

### Test Merges

(Not relevant but)

#7406: 'Super Rage Gives Ravager Bloodborne/Life Leech Slashing' by Surrealaser at commit 03c5d2d8df
#7442: 'Adds the HMG to marine vendors, HSG rebalance.' by Pariah919 at commit ca3c4e6be3
#7515: 'Fixes the NTOS crash and round start text for MD and CMO' by Hekzder at commit d08ec53791

### Reproduction Steps

1. Go to the Follow/Orbit menu
2. Hit the refresh button, it does not work.
3. 
image

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Reworks the orbit UI to be quite a bit more interactive. Could probably use TM'd, it's hard to test every icon and case.

This PR adds:
- Health-colored buttons
- Xeno nickname buttons
- Section colors
- Collapsible sections
- Job info
- Job icons
- Search by caste/job/squad
- Fixes #7521

<details>
<summary>Check it out</summary>
some basic functionality. searching by job, caste. tooltips.

WNnKjVaNBh

buttons are colored by health

Screenshot 2022-10-27 102740

</details>

## Why It's Good For The Game
Better observer experience. Adds funny names, icons, health bars.

## Changelog
:cl:
qol: Reworks the Orbit UI to be much more interactive. Colors buttons on health, collapsible sections, job icons, and more.
fix: Orbit ui refresh button should be working now
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
Nice

File: code/modules/mob/dead/observer/orbit.dm [hunk 1]
```diff
@@ -72,35 +58,59 @@
...
-			if(ismarinejob(H.job))
+			continue
+
+		if(ishuman(mob_poi))
+			var/mob/living/carbon/human/human = poi
+			if(ismarinejob(human.job))
+				var/datum/job/marine_job = human.job
+				serialized["job"] = marine_job?.title
+				var/icon/orbit_icon = icon('icons/UI_icons/map_blips.dmi', initial(marine_job?.minimap_icon))
+				serialized["icon"] = icon2base64(orbit_icon)
```
Comment (TiviPlus) [inline code/modules/mob/dead/observer/orbit.dm:104]: This seems very wasteful, couldnt you just cache this easily?

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Ui is very nice

**Comment by lKiseki** [comment]
Beautiful update, love what you did here.

Just here to report a bug, though;
image
There's currently a bug where the orbit UI will show all of these icons in the Marines tab.

**Comment by jlsnow301** [comment]
There's currently a bug where the orbit UI will show all of these icons in the Marines tab.

lol, nice, on it right now

**Comment by comfyorange** [comment]
#### Test Merge Removed

**Comment by jlsnow301** [comment]
Should be safe to put on TM again.

**Comment by Edelor68** [comment]
Does this fix #11049?

**Comment by jlsnow301** [comment]
> Does this fix #11049?

i don't think so, the underlying issue there seems that they get categorized as NPCs or something (mind == null). 
i will test it and report the results.

**Comment by XSlayer300** [comment]
Disabling Auto-observe does not unobserve the selected target. Whereas before, you disable auto-observe to disable spectating the person and their inventory.

You can still disable spectating the person by orbiting another target while auto-observe is off, which is weird.

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 74ddc18f3425d2a813310b95d0a2e0f96909bce9
Pull Request: e3d723deb37eb08ac9d44567914a3a87023f4f1b
Server: 0630093ff9a7e29366983fccc1d30b0735259a83

##### Comment

**Comment by jlsnow301** [comment]
> You can still disable spectating the person by orbiting another target while auto-observe is off, which is weird.

I turned it to a client state, I'm not totally against just reverting this to fix it (I just don't like server states)

**Comment by comfyorange** [comment]
#### Test Merge Removed

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**